### DISCUSSION TITLE
Installing PDM requires crypto libraries that also depend on Rust

### DISCUSSION THREAD

**Discussion by andrewebdev** [pdm-project/pdm#426] [Q&A] [answered]
I'm trying to usd PDM in my CI/CD process on Gitlab. It's using the `python:3.8-alpine` image.

Eventually it tries to install the cryptography library (is this required for PDM?), and this then fails stating that it depends on the Rust library. This feels like a very large overhead just to have a package manager. Is this intended or is something else happinging here? It's slowing down the CI/CD pipeline immensely.

My CI/CD used to use just `pip` and didn't require crypography or rust.

**Comment by pawamoy** [comment, marked as answer]
`cryptography` is a dependency of `keyring`, which is used for [this](https://github.com/pdm-project/pdm/blob/e3cda1d039a4fd06a27cc58d8e06e1cc589f0605/docs/docs/usage/project.md#hide-the-credentials-from-pyprojecttoml) in PDM. Maybe this could be made optional.

**Reply by andrewebdev** [reply]
aaah ok that makes sense to me now. In this case I think that I'll just use normal pip for my CI/CD pipeline, since it's an edge case just to optimize that step (It's just static files being built by Sphinx). I can still use PDM in my dev environment.
